src/Handlers/NetworkTestHandler.ts

```diff
@@ -54,7 +55,14 @@ export class NetworkTestHandler
           chunk.indexOf(ReadlinePlatform.HOST_OR_IP_QUESTION) > -1 &&
           config.type === NetworkTestType.TRACEROUTE
         ) {
-          child.stdin.write(`${config.host}${EOL}`);
+          let hostOrIP = '';
+          try {
+            hostOrIP = new URL(config.host).host;
+          } catch (error) {
+            hostOrIP = config.host;
+          }
+
+          child.stdin.write(`${hostOrIP}${EOL}`);
```

Comment:
Well, well... `host !=== hostname`. If you decided to rework API to accept `hostname` instead of `url`, you should rename the field name as well. Otherwise, you should parse `url` to obtain `hostname` as I mentioned above:

```ts
let hostname = '';
try {
  ({ hostname} = new URL(config.url));
} catch {
  hostname = config.url;
}

child.stdin.write(`${hostname}${EOL}`);
```

or

```ts
let hostname = '';
try {
  ({ hostname} = new URL(`https://${config.hostname}`));
} catch {
  hostname = config.hostname;
}

child.stdin.write(`${hostname}${EOL}`);
```

Reply:
I'm sorry, but I understand why we make URL from host.
We must got host or IP from the backend not URL.

Reply:
I'm confused. It looks like you read my comments diagonally. See difference between `host` and `hostname`